### PR TITLE
Integrate from Piper for C++, Java, and Python

### PR DESCRIPTION
**Pull request by haberman**
Important: this integration must be merged, not squashed!

This is a continuation of https://github.com/protocolbuffers/protobuf/pull/9461

### CODE REVIEW

File: python/setup.py [hunk 1]
```diff
@@ -301,12 +301,11 @@ def get_option_from_sys_argv(option_str):
       classifiers=[
           "Programming Language :: Python",
           "Programming Language :: Python :: 3",
```
Comment (acozzette) [inline python/setup.py:303]: I think some setup.py changes got lost again here. The good news is that after this PR is merged, the problem should be solved going forward.

File: java/core/src/main/java/com/google/protobuf/ByteString.java [hunk 2]
```diff
@@ -279,8 +316,13 @@ public int compare(ByteString former, ByteString latter) {
       };
 
   /**
+<<<<<<< HEAD
    * Returns a {@link Comparator} which compares {@link ByteString}-s lexicographically
    * as sequences of unsigned byte values between 0 and 255, inclusive.
+=======
+   * Returns a {@link Comparator} which compares {@link ByteString}-s lexicographically as sequences
```
Comment (acozzette) [inline java/core/src/main/java/com/google/protobuf/ByteString.java:323]: I think the resolution this conflict should be something like this:
```
   * Returns a {@link Comparator} which compares {@link ByteString}-s lexicographically as sequences
   * of unsigned byte values between 0 and 255, inclusive.
```

**Comment by acozzette** [comment]
I saw that there are a few more merge conflicts that need to be resolved. If you just go to https://github.com/protocolbuffers/protobuf/pull/9479.patch you can find them by searching for `<<<<`.

File: python/setup.py (hunk 1 above)
Comment (haberman) [reply python/setup.py]: I fixed this.

File: java/core/src/main/java/com/google/protobuf/ByteString.java (hunk 2 above)
Comment (haberman) [reply java/core/src/main/java/com/google/protobuf/ByteString.java:319]: Fixed.

**Review by acozzette — APPROVED** [review]

**Pull request merged**